docs/explore_api/templating.md

```diff
@@ -3,307 +3,384 @@
-The labels we just assigned will now be treated as names of variables whose values
-we can programmatically modify using Python. 
+```python
+template_manager = ...
+template_manager.render_file('objects/obj1.def.jinja',
+                             target_file='my_objects/sphere.def',
+                             radius=10)
+```
+
+```console
+$ tree my_rendered_files
+my_rendered_files
+└── my_objects
+    └── sphere.def
+```
+
+### Adding external resources
 
-Saving our template file and the `unit_box.obj` file in a new folder named `splishsplash-template`
-we can now run the same simulation with different properties we set programmatically.
+In addition to rendering template files, the `TemplateManager` class also provides
+mechanisms to add external files to the destination directory. This method is useful
+when you need to copy files that are not part of the template directory but are
+required for the simulation. The `copy_dir` and `copy_file` methods allow you to
+copy entire directories or individual files to the destination directory, respectively.
+
+In the following examples, we add external resources to the destination directory,
+first by adding an entire directory and then by adding a single file:
 
-This is how we would do it for a viscous fluid:
 
 ```python
-import inductiva
+template_manager = ...
+template_manager.copy_dir('/path/to/external_resources')
+template_manager.copy_file('./external_file.txt')
+```
+
+```console
+$ tree my_rendered_files
+my_rendered_files
+├── external_file.txt
+└── external_resources
+    ├── ...
+```
+
+Similarly to the `render_dir` and `render_file` methods, you can specify the
+destination name of the copied directory or file:
 
-# Launch a machine group with a c2-standard-30
-machine_group = inductiva.resources.MachineGroup(machine_type="c2-standard-30")
-machine_group.start()
 
-# Download the input files mentioned above
-input_dir = inductiva.utils.download_from_url(
-    "https://storage.googleapis.com/inductiva-api-demo-files/"
-    "splishsplash-template-example.zip", unzip=True)
+```python
+template_manager = ...
+template_manager.copy_dir('/path/to/external_resources',
+                          'resources/copied_resources')
+template_manager.copy_file('./external_file.txt',
+                           'copied_file.txt')
+```
 
-# Set the new fluid properties at room temperature
-density = 1500 # kg/m^3
-kinematic_viscosity = 1 # m^2/s
+```console
+$ tree my_rendered_files
+my_rendered_files
+├── copied_file.txt
+└── resources
+    └── copied_resources
+        ├── ...
+```
 
-# Initialize the simulator and run the simulation
-splishsplash = inductiva.simulators.SplishSplash()
 
-# Initialize the templating manager
-template_manager = inductiva.TemplateManager(template_dir=input_dir,
-                                    root_dir="splishsplash-viscous")
+The `TemplateManager` class also provides a method to add external files to the
```

Comment:
This lines ends with nothing below it or am I missing something?

Reply:
It was supposed to be deleted. Addressed in [1b4d5b1](https://github.com/inductiva/inductiva/pull/1249/commits/1b4d5b106ec2d2381eb4242fdae6e06b662279e5).